### ISSUE TITLE
Implement HTTP proxy support

### ISSUE THREAD

**Issue by clue** [reactphp/http-client#44]

**Comment by clue** [comment]
I've started looking into this a while back and it actually worked out quite good, but I was never quite happy with its API. As such, I'm looking into adding PSR-7 support via #41 first :+1: 

In the meantime, these proxy servers are already supported:

* SOCKS: https://github.com/clue/php-socks-react
* HTTP CONNECT: https://github.com/clue/php-http-proxy-react

**Comment by enrico69** [comment]
Hi
So is there any way to connect through a proxy?

**Comment by clue** [comment]
@enrico69 Absolutely! Have you tried any of the above links? Simply pass a custom connector instance to the `Client` and everything should work :+1: 

Note that plain HTTP proxies are currently not supported yet, hence why this ticket is still open.

**Comment by enrico69** [comment]
@clue 
Hi, and thanks for your answer.

However is this solution still up to date?
When I do a composer require, it fails because the current version of ReactPHP is not the good one.
Not compatible I guess, beacause if I try a manual installation, the class ProxyConnector uses classes wich seems to no longer exit, as _React\Socket\ConnectorInterface_ for example. So unable to try your solution :(

**Comment by enrico69** [comment]
OK I finally managed to do it.
I completely removed ReactPHP with a _composer remove react/react_
Then I installed the proper libraries:

_composer require clue/http-proxy-react:^1.1
composer require react/http-client_

Thanks for your work
